File: time2secs/time2secs.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func time2secs(str string) int {

	var (
		seconds     int
		s           int
		time        []string
		conversions [3]int
	)

	conversions = [3]int{1, 60, 3600}
	time = strings.Split(str, ":")
	reverse(time)

	for i, t := range time {
		s, _ = strconv.Atoi(t)
		seconds += s * conversions[i]
	}

	return seconds
}

func reverse(s []string) {
	for i := len(s)/2 - 1; i >= 0; i-- {
		j := len(s) - 1 - i
		s[i], s[j] = s[j], s[i]
	}
}

func main() {

	if len(os.Args) == 2 {
		fmt.Println("===", time2secs(os.Args[1]), "seconds ===")
	} else {
		fmt.Println("[Incorrect input]")
		fmt.Println("example: ./time2secs 12:34:56")
	}
}

File: time2secs/time2secs.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int count(char time[], char delim) {
    // count the occurrences of the delimiter

    int occurrences = 0;
    for (int i = 0; time[i] != '\0'; i++)
        if (time[i] == delim)
            occurrences++;

    return occurrences;

}
int time_to_seconds(char *time) {
    // convert a time string to seconds

    char *token;
    int t[] = {0, 0, 0};
    int i = count(time, ':');

    // populate array in reverse
    // depending on number of indices
    token = strtok(time, ":");
    t[i--] = atoi(token);

    while ((token = strtok(NULL, ":")) != NULL)
        t[i--] = atoi(token);

    return t[0] + t[1]*60 + t[2]*3600;
}

int main(int argc, char *args[]) {
    if (argc == 2) {
        printf("=== %d seconds ===\n", time_to_seconds(args[1]));
    }
    else {
        puts("Please enter a command line arg");
        puts("HH:MM:SS");
        puts("Example:");
        puts("  time2secs 12:34:56");
    }
    return 0;
}

